astropy/units/core.py

```diff
@@ -807,15 +807,26 @@ def __pow__(self, p: Real) -> CompositeUnit:
                 ) from None
             return CompositeUnit(1, [self], [sanitize_power(p)], _error_check=False)
 
-    def __truediv__(self, m):
-        if isinstance(m, (bytes, str)):
-            m = Unit(m)
+    @staticmethod
+    def _warn_about_operation_with_deprecated_type(op: str, other: bytes | str) -> None:
```

Comment:
I implemented this as a static method instead of a module level function so that it would be close to where it is used. It's a temporary private method, so it doesn't matter too much where put it.